test/AspNetCoreModule.Test/Framework/InitializeTestMachine.cs

```diff
@@ -323,30 +446,40 @@ private void PreparePrivateANCMFiles()
+                        //
 
-                        string from = Path.Combine(outputPath, "x64", "aspnetcore.dll");
-                        TestUtility.FileCopy(from, FullIisAspnetcore_path, overWrite:true, ignoreExceptionWhileDeletingExistingFile:false);
-                        TestUtility.FileCopy(from, IisExpressAspnetcore_path, overWrite: true, ignoreExceptionWhileDeletingExistingFile: false);
+                        if (isIISReady)
+                        {
+                            TestUtility.ResetHelper(ResetHelperMode.KillWorkerProcess);
+                            TestUtility.ResetHelper(ResetHelperMode.StopW3svcStartW3svc);
+                            Thread.Sleep(1000);
+                        }
 
-                        // NOTE: schema file can't be overwritten, if there is any schema change, that should be updated manually
-                        from = Path.Combine(outputPath, "x64", "aspnetcore_schema.xml");
-                        TestUtility.FileCopy(from, FullIisAspnetcoreSchema_path, overWrite: false, ignoreExceptionWhileDeletingExistingFile: false);
-                        TestUtility.FileCopy(from, IisExpressAspnetcoreSchema_path, overWrite: false, ignoreExceptionWhileDeletingExistingFile: false);
+                        string from = Path.Combine(outputPath, "x64", "aspnetcore.dll");
+                        if (isIISReady)
+                        {
+                            // Copy private file on Inetsrv directory
+                            TestUtility.FileCopy(from, FullIisAspnetcore_path, overWrite: true, ignoreExceptionWhileDeletingExistingFile: false);
+                        }
 
+                        // Initialize IisExpressAspnetcore_path
+                        IisExpressAspnetcore_path = from;
+                                                
                         if (TestUtility.IsOSAmd64)
                         {
-                            from = Path.Combine(outputPath, "Win32", "aspnetcore.dll");
-                            TestUtility.FileCopy(from, FullIisAspnetcore_X86_path, overWrite: true, ignoreExceptionWhileDeletingExistingFile: false);
-                            TestUtility.FileCopy(from, IisExpressAspnetcore_X86_path, overWrite: true, ignoreExceptionWhileDeletingExistingFile: false);
+                            string from_32bit = Path.Combine(outputPath, "Win32", "aspnetcore.dll");
 
-                            // NOTE: schema file can't be overwritten, if there is any schema change, that should be updated manually
-                            from = Path.Combine(outputPath, "Win32", "aspnetcore_schema.xml");
-                            TestUtility.FileCopy(from, IisExpressAspnetcoreSchema_X86_path, overWrite: false, ignoreExceptionWhileDeletingExistingFile: false);
-                        }
+                            if (isIISReady)
```

Comment:
There are a lot of checks for that bool. Consider separating into two methods rather than checking it multiple times.

Reply:
Fixed.